frontend/src/components/data-table/column-formatting/applyFormat.ts

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import { DataType } from "@/core/kernel/messages";
+import { type FormatOption } from "./types";
+import { prettyNumber, prettyScientificNumber } from "@/utils/numbers";
+
+// Apply formatting to a value given a format and data type
+export const applyFormat = (
+  value: unknown,
+  format: FormatOption,
+  dtype: DataType | undefined,
+) => {
+  switch (dtype) {
+    case "date": {
+      const date = new Date(value as string);
+      switch (format) {
+        case "MM/DD/YY":
+          return date.toLocaleDateString("en-US");
+        case "DD/MM/YY":
+          return date.toLocaleDateString("en-GB");
+        case "YYYY-MM-DD":
+          return date.toISOString().split("T")[0];
+        case "HH:mm:ss":
+          return date.toTimeString().split(" ")[0];
+        default:
+          return value;
+      }
+    }
+    case "number": {
+      const num = Number.parseFloat(value as string);
+      switch (format) {
+        case "auto":
+          return prettyNumber(num);
+        case "percent":
+          return `${(num * 100).toFixed(2)}%`;
+        case "scientific":
+          return prettyScientificNumber(num);
+        case "int":
+          return num.toFixed(0);
+        case "0.0":
+          return num.toFixed(1);
+        case "0.00":
+          return num.toFixed(2);
+        default:
+          return value;
+      }
+    }
+    case "string":
+      switch (format) {
+        case "uppercase":
+          return (value as string).toUpperCase();
+        case "lowercase":
+          return (value as string).toLowerCase();
+        case "capitalize":
+          return (
+            (value as string).charAt(0).toUpperCase() +
+            (value as string).slice(1)
+          );
+        case "title":
+          return (value as string).replaceAll(/\b\w/g, (char) =>
+            char.toUpperCase(),
+          );
+        case "sentence":
+          return (
+            (value as string).charAt(0).toUpperCase() +
+            (value as string).slice(1)
+          );
+        default:
+          return value;
+      }
+    case "boolean":
+      switch (format) {
+        case "true/false":
+          return (value as boolean).toString();
```

Comment:
this is probably the default right? should we just remove this since it wont change anything?